### PR TITLE
Generalized the response writer to io.Writer

### PR DESCRIPTION
**Pull request by theSuess**
This adds support for the echo framework and many other custom web frameworks

### CODE REVIEW

**Comment by unrolled** [comment]
This looks good! Can you fix the broken test as well? In `engine_integration_test.go` the method needs to be updated to reflect `w io.Writer`:
```go
func (a Amber) Render(w http.ResponseWriter, v interface{}) error {
	a.Head.Write(w)
	return a.Template.Execute(w, v)
}
```

**Comment by theSuess** [comment]
Sorry, I didn't see the integration tests.

I fixed the tests and added an echo example to the readme

File: README.md [hunk 1]
```diff
@@ -492,3 +492,40 @@ func main() {
     router.Run()
 }
 ~~~
+
+### [Echo](https://github.com/labstack/echo)
```
Comment (unrolled) [inline README.md]: This can actually replace the one at the top of the example list (line 361).

**Comment by theSuess** [comment]
All done!

**Comment by unrolled** [comment]
Awesome, thanks!!

**Pull request merged**